tests/ConstantTest.php

```diff
@@ -314,6 +314,27 @@ class Foo {
                 ',
                 'assertions' => ['$foo' => 'string'],
             ],
+            'dynamicClassConstFetch' => [
+                '<?php
+                    class Foo
+                    {
+                        public const BAR = "bar";
+                    }
+
+                    $foo = new Foo();
+                    $_trace = $foo::BAR;',
+                'assertions' => ['$_trace===' => '"bar"'],
+            ],
+            'dynamicClassConstFetchClassString' => [
+                '<?php
+                    class C {
+                        public const CC = 1;
+                    }
+
+                    $c = C::class;
+                    $d = $c::CC;',
+                'assertions' => ['$d===' => '1'],
+            ],
```

Comment:
It would be great to add couple more tests, to showcase the difference between `can_be_inheritance=true` (or whatever we end up calling it) and the opposite of that.

Reply:
done, I added a case with final and a case without final nor a definite_class to show `mixed` result